### PR TITLE
add augmentor support

### PR DESCRIPTION
**Pull request by fumihwh**

### CODE REVIEW

**Comment by hadusam** [comment]
run test

**Comment by hadusam** [comment]
run test

**Comment by hadusam** [comment]
run test

**Comment by hadusam** [comment]
run dlk test

**Comment by hadusam** [comment]
run test

**Comment by hadusam** [comment]
run test

**Comment by lm-jira** [comment]
Library for FPGA can be generated from this code without any problem.
The library can also be implemented without problem

**Comment by hadusam** [comment]
@lm-jira Thank you for your checking!
@ruimashita Please review this PR. 👍

File: blueoil/blueoil_init.py [hunk 1]
```diff
@@ -13,11 +13,18 @@
...
 # limitations under the License.
 # =============================================================================
+import inspect
 import re
+import sys
 
 import whaaaaat
 from jinja2 import Environment, FileSystemLoader
 
+sys.path.insert(0, "lmnet")
```
Comment (ruimashita) [inline blueoil/blueoil_init.py]: May not need this line. I will fix it.

File: blueoil_test.sh [hunk 2]
```diff
@@ -88,8 +88,10 @@ function init_test(){
     TASK_TYPE_NUMBER=$2
     NETWORK_NUMBER=$4
     DATASET_FORMAT_NUMBER=$5
-    TRAINING_DATASET_PATH=$6
-    VALIDATION_DATASET_PATH=$7
+    ENABLE_DATA_AUGMENTATION=$6
+    CHOOSE_TOP_TWO_AUG=$7
```
Comment (ruimashita) [inline blueoil_test.sh]: Didn't use this var `CHOOSE_TOP_TWO_AU`. need to delete this line.

**Comment by ruimashita** [comment]
data_augmentor page is empty content
https://deploy-preview-2--blueoil.netlify.com/reference/data_augmentor.html

**Comment by fumihwh** [comment]
No idea about that. It works good locally.

**Comment by ruimashita** [comment]
@fumihwh OK. I will do the document things 👍

**Comment by ruimashita** [comment]
@lm-jira 
When I select `RandomErasingForDetection` and `./blueoil.sh train` the error comes.

```
yaml.scanner.ScannerError: mapping values are not allowed here
  in "/home/blueoil/config/aa.yml", line 34, column 29
ERROR: Training exited with a non-zero status
(.venv) #( 11/02/18@ 5:12PM )( wakisaka@leapmind-se
```

That's way the generated yaml has broken format.
```
# supported task types are 'classification', 'object_detection' and 'semantic_segmentation'.
task_type: classification

network_name: LmnetV1Quantize

dataset:
  format: Caltech101
  train_path:
  test_path:

trainer:
  batch_size: 10
  epochs: 100

common:
  image_size:
    - 128  # height
    - 128  # width

  # set pretrain model name. currently, this feature is not supported, always ignored.
  pretrain_model: false

  # enable dataset prefetch, set false if weired problem happens
  dataset_prefetch: true

  data_augmentation:
    - RandomErasingForDetection:
      - probability: 0.5- sl: 0.02- sh: 0.2- r1: 0.3- content_type: mean- mean: [125, 122, 114]- i_a: True- o_a: True
```

What kind of manually test, did you do ??

**Comment by lm-jira** [comment]
I trained the model without editing anything in the config file.

**Pull request closed without merging**